src/rebalancers/Rebalancer.sol

```diff
@@ -0,0 +1,492 @@
+/**
+ * Created by Pragma Labs
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+pragma solidity 0.8.22;
+
+import { ActionData, IActionBase } from "../../lib/accounts-v2/src/interfaces/IActionBase.sol";
+import { ArcadiaLogic } from "./libraries/ArcadiaLogic.sol";
+import { BurnLogic } from "./libraries/BurnLogic.sol";
+import { ERC20, SafeTransferLib } from "../../lib/accounts-v2/lib/solmate/src/utils/SafeTransferLib.sol";
+import { FeeLogic } from "./libraries/FeeLogic.sol";
+import { FixedPointMathLib } from "../../lib/accounts-v2/lib/solmate/src/utils/FixedPointMathLib.sol";
+import { FullMath } from "../../lib/accounts-v2/src/asset-modules/UniswapV3/libraries/FullMath.sol";
+import { IAccount } from "./interfaces/IAccount.sol";
+import { IPool } from "./interfaces/IPool.sol";
+import { IPositionManager } from "./interfaces/IPositionManager.sol";
+import { IStrategyHook } from "./interfaces/IStrategyHook.sol";
+import { MintLogic } from "./libraries/MintLogic.sol";
+import { PricingLogic } from "./libraries/PricingLogic.sol";
+import { RebalanceLogic } from "./libraries/RebalanceLogic.sol";
+import { SafeApprove } from "./libraries/SafeApprove.sol";
+import { SlipstreamLogic } from "./libraries/SlipstreamLogic.sol";
+import { StakedSlipstreamLogic } from "./libraries/StakedSlipstreamLogic.sol";
+import { SwapLogic } from "./libraries/SwapLogic.sol";
+import { SqrtPriceMath } from "./libraries/uniswap-v3/SqrtPriceMath.sol";
+import { TickMath } from "../../lib/accounts-v2/src/asset-modules/UniswapV3/libraries/TickMath.sol";
+import { UniswapV3Logic } from "./libraries/UniswapV3Logic.sol";
+
+/**
+ * @title Permissioned rebalancer for Uniswap V3 and Slipstream Liquidity Positions.
+ * @notice The Rebalancer will act as an Asset Manager for Arcadia Accounts.
+ * It will allow third parties to trigger the rebalancing functionality for a Liquidity Position in the Account.
+ * The owner of an Arcadia Account should set an initiator via setAccountInfo() that will be permisionned to rebalance
+ * all Liquidity Positions held in that Account.
+ * @dev The contract prevents frontrunning/sandwiching by comparing the actual pool price with a pool price calculated from trusted
+ * price feeds (oracles). The tolerance in terms of price deviation is specific to the initiator but limited by a global MAX_TOLERANCE.
+ * @dev The contract guarantees a very limited loss of principal with each rebalance by enforcing a minimum amount of liquidity that must be added,
```

Comment:
I wouldn't use "loss" of principal here, as leftovers are sent back to the Account

Reply:
or if related to slippage then precise minimum slippage (which you would always have, not a loss)